tools/topology/topology1/CMakeLists.txt

```diff
@@ -0,0 +1,216 @@
+	"sof-cml-demux-rt5682-max98357a\;sof-cml-demux-rt5682-max98357a\;-DPLATFORM=cml"
+	"sof-cml-rt1011-rt5682\;sof-cml-rt1011-rt5682\;-DPLATFORM=cml\;-DPPROC=volume"
+	"sof-tgl-nocodec\;sof-tgl-nocodec"
+	"sof-tgl-rt711-rt1308\;sof-tgl-rt711-rt1308-2ch\;-DCHANNELS=2\;-DEXT_AMP\;-DDMICPROC_FILTER1=eq_iir_coef_highpass_40hz_20db_48khz.m4\;-DDMIC16KPROC_FILTER1=eq_iir_coef_highpass_40hz_20db_16khz.m4\;-DPLATFORM=tgl"
+	"sof-tgl-rt711-rt1308\;sof-tgl-rt711-rt1308-4ch\;-DCHANNELS=4\;-DEXT_AMP\;-DDMICPROC_FILTER1=eq_iir_coef_highpass_40hz_20db_48khz.m4\;-DDMIC16KPROC_FILTER1=eq_iir_coef_highpass_40hz_20db_16khz.m4\;-DPLATFORM=tgl"
+	"sof-tgl-rt711-rt1308\;sof-adl-rt711-4ch\;-DCHANNELS=4\;-DDMICPROC_FILTER1=eq_iir_coef_highpass_40hz_20db_48khz.m4\;-DDMIC16KPROC_FILTER1=eq_iir_coef_highpass_40hz_20db_16khz.m4\;-DPLATFORM=adl"
+	"sof-tgl-rt711-rt1308\;sof-adl-rt711\;-DCHANNELS=0\;-DPLATFORM=adl"
+	"sof-tgl-nocodec\;sof-ehl-nocodec\;-DDYNAMIC=1"
+	"sof-ehl-rt5660\;sof-ehl-rt5660\;-DHDMI=1"
+	"sof-ehl-rt5660\;sof-ehl-rt5660-nohdmi"
+	"sof-imx8-nocodec\;sof-imx8-nocodec"
+	"sof-imx8-cs42888\;sof-imx8-cs42888"
+	"sof-imx8-nocodec-sai\;sof-imx8-nocodec-sai"
+	"sof-imx8-wm8960\;sof-imx8-wm8960"
+	"sof-imx8-wm8960-mixer\;sof-imx8-wm8960-mixer"
+	"sof-imx8-wm8960-kwd\;sof-imx8-wm8960-kwd"
+	"sof-imx8-cs42888-mixer\;sof-imx8-cs42888-mixer"
+	"sof-imx8mp-wm8960-mixer\;sof-imx8mp-wm8960-mixer"
+	"sof-imx8mp-wm8960-kwd\;sof-imx8mp-wm8960-kwd"
+	"sof-tgl-max98357a-rt5682\;sof-tgl-max98357a-rt5682\;-DCODEC=MAX98357A\;-DFMT=s16le"
+	"sof-tgl-max98357a-rt5682\;sof-tgl-max98357a-rt5682-pdm1\;-DCODEC=MAX98357A\;-DFMT=s16le\;-DDMIC_DAI_LINK_16k_PDM=STEREO_PDM1"
+	"sof-tgl-max98357a-rt5682\;sof-tgl-rt1011-rt5682\;-DCODEC=RT1011\;-DFMT=s24le"
+	"sof-tgl-max98373-rt5682\;sof-tgl-max98373-rt5682\;-DAMP_SSP=1"
+	"sof-tgl-max98373-rt5682\;sof-tgl-max98373-rt5682-igonr\;-DAMP_SSP=1\;-DIGO"
+	"sof-tgl-max98373-rt5682\;sof-tgl-max98373-rt5682-xperi\;-DAMP_SSP=1\;-DINCLUDE_IIR_EQ=1"
+	"sof-tgl-max98373-rt5682\;sof-tgl-rt5682-ssp0-max98373-ssp2\;-DAMP_SSP=2"
+	"sof-tgl-max98373-rt5682\;sof-tgl-rt5682-ssp0-max98373-ssp2-xperi\;-DAMP_SSP=2\;-DINCLUDE_IIR_EQ=1"
+	"sof-tgl-sdw-max98373-rt5682\;sof-tgl-sdw-max98373-rt5682\;-DCHANNELS=4\;-DPLATFORM=tgl"
+	"sof-tgl-sdw-max98373-rt5682\;sof-adl-sdw-max98373-rt5682\;-DCHANNELS=4\;-DPLATFORM=adl"
+	"sof-jsl-da7219\;sof-jsl-da7219\;-DPLATFORM=jsl"
+	"sof-jsl-da7219\;sof-jsl-da7219-mx98360a\;-DPLATFORM=jsl-dedede"
+	"sof-imx8mp-wm8960\;sof-imx8mp-wm8960"
+	"sof-smart-amplifier-nocodec\;sof-smart-amplifier-nocodec"
+	"sof-jsl-rt5682\;sof-jsl-rt5682-rt1015\;-DPLATFORM=jsl-rt1015"
+	"sof-jsl-rt5682\;sof-jsl-rt5682-rt1015-xperi\;-DPLATFORM=jsl-rt1015\;-DINCLUDE_IIR_EQ=1"
+	"sof-jsl-rt5682\;sof-jsl-rt5682-mx98360a\;-DPLATFORM=jsl-dedede"
+)
+
+add_custom_target(topologies1_output ALL)
+add_custom_target(topologies1 ALL
+    COMMAND ${CMAKE_COMMAND} -E copy_directory . ${SOF_TOPOLOGY_BINARY_DIRECTORY}
+    DEPENDS topologies1_output
```

Comment:
I find the names `topologies1_output` and `topologies1` not very clear. How about:
```
add_custom_target(topologies1 ALL)
add_custom_target(topologies1_copy ALL
   COMMAND copy_directory
```
End users should rarely ever invoke "make topologies1" directly. The very few who do should know what they're doing.

Or even simpler, have you tried to combine these two into a single `add_custom_target`? Shorter code and naming issue goes away

```
add_custom(topologies1 ALL
          COMMAND copy_directory...
...
)
...
add_dependencies(topologies1 topology1_${output})
```
The dependencies are obviously built before the COMMAND is run.